### PR TITLE
Update vault get-secrets to support v2 KV Secrets Engine

### LINKED ISSUE #17: Vault - support KV Secrets Engine V2
`dockcmd vault get-secrets` fails with `Could not convert vault response [%s][%s] to string` ([code link](https://github.com/boxboat/dockcmd/blob/develop/cmd/vault.go#L100)) if the secret was created with  [KV V2](https://www.vaultproject.io/api/secret/kv/kv-v2.html) (versioned secret).

V2 secrets need to read values from `data.data.key` instead of `data.key`. Either allowing the user to specify the version type in the template, or even better, automatically attempting to parse a level deeper if it fails at the `data.key` check would allow users to read both v1 and v2 secrets

### PR DESCRIPTION
**Pull request by boxboatmatt**
Signed-off-by: Matthew DeVenny <matt@boxboat.com>

Adds support for `v2` KV Secrets Engine
Closes #17

### CODE REVIEW

**Comment by jcantosz** [comment]
Nice! This is very useful functionality!
I compiled and tested. This is working for me on v1 and v2 secrets (when adding /data/)

I noticed that `vault kv get` seems to detect the secret version automatically. 
I found this in the command [isKVv2(path, client)](https://github.com/hashicorp/vault/blob/7aa1ffa92ee61b977efad1488b8f309b1e2136df/command/kv_get.go#L94) and [this definition](https://github.com/hashicorp/vault/blob/0c9b0117093df05bd7539325c500ec3fbdff982d/command/kv_helpers.go#L99). Haven't dug into the implementation, but would it be possible/make sense to import and use this functionality?

**Comment by boxboatmatt** [comment]
> Nice! This is very useful functionality!
> I compiled and tested. This is working for me on v1 and v2 secrets (when adding /data/)
> 
> ~I noticed that `vault kv get` seems to detect the secret version automatically (automatically adds /data/). I found this in the command [isKVv2(path, client)](https://github.com/hashicorp/vault/blob/7aa1ffa92ee61b977efad1488b8f309b1e2136df/command/kv_get.go#L94) and [this definition](https://github.com/hashicorp/vault/blob/0c9b0117093df05bd7539325c500ec3fbdff982d/command/kv_helpers.go#L99). Haven't dug into the implementation, but would it be possible/make sense to import and use this functionality?~
> Scratch that, not part of the api. Probably best not to depend on it

Just looked at it... we could copy the 2 methods they wrote to check against it. I looked for a way to do just this but was only looking at the API

File: cmd/vault_kv_helpers.go [hunk 1]
```diff
@@ -0,0 +1,85 @@
+package cmd
```
Comment (jcantosz) [inline cmd/vault_kv_helpers.go:1]: Original code licensed under MPL https://github.com/hashicorp/vault/blob/master/LICENSE. I believe we need to add a license header here: 
https://www.mozilla.org/en-US/MPL/headers/

Comment (boxboatmatt) [reply cmd/vault_kv_helpers.go:6]: License added

**Pull request merged**